File: 1590.make-sum-divisible-by-p/solution.go
func minSubarray(nums []int, p int) int {
    remainder := 0
    for _, n := range nums {
        remainder = (remainder + n) % p
    }
    if remainder == 0 {
        return 0
    }
    modMap := map[int]int{0: -1}
    currentSum := 0
    minLen := len(nums)
    for i, n := range nums {
        currentSum = (currentSum + n) % p
        needed := (currentSum - remainder + p) % p
        if j, exists := modMap[needed]; exists {
            minLen = min(minLen, i - j)
        }
        modMap[currentSum] = i
    }
    if minLen == len(nums) {
        return -1
    }
    return minLen
}

File: 1590.make-sum-divisible-by-p/solution.kt
class Solution {
    fun minSubarray(nums: IntArray, p: Int): Int {
        val remainder = nums.fold(0) { acc, num -> (acc + num) % p }

        if (remainder == 0) {
            return 0
        }

        val lastOccurrence = mutableMapOf(0 to -1)
        var currentSum = 0
        var res = nums.size

        nums.forEachIndexed { i, num ->
            currentSum = (currentSum + num) % p
            val needed = (currentSum - remainder + p) % p
            lastOccurrence[needed]?.let { j -> res = minOf(res, i - j) }
            lastOccurrence[currentSum] = i
        }

        return if (res == nums.size) -1 else res
    }
}